lib/cc/cli/analyze.rb

```diff
@@ -72,6 +64,27 @@ def process_args
       def formatter
         @formatter ||= Formatters::PlainTextFormatter.new(filesystem)
       end
+
+      def disable_all_engines!
+        unless engines_disabled
```

Comment:
Do you expect this to be called more than once? Wondering why we need the `engines_disabled` check.

Reply:
If you pass `-e foo -e bar`, you only want to disable all the engines one time, otherwise you'll *only* end up with `bar` enabled.

This logic was inline in `process_args`: I extracted it to a method, and kept the once-only guard here since otherwise it didn't seem worth extracting.

Reply:
Ahh, that makes sense 👍  I didn't realize `-e` was accepted multiple times.